api/python/quilt3/api.py

```diff
@@ -196,17 +197,17 @@ def _config(*catalog_url, **config_values):
         else:
             config_template = read_yaml(CONFIG_TEMPLATE)
             write_yaml(config_template, CONFIG_PATH, keep_backup=True)
-        return QuiltConfig(CONFIG_PATH, config_template)
-
-    # Use local configuration (or defaults)
-    local_config = load_config()
-
-    # Write to config if needed
-    if config_values:
+        local_config = config_template
+    # Create a custom config with the passed-in values only
+    elif config_values:
+        local_config = load_config()
         config_values = QuiltConfig('', config_values)  # Does some validation/scrubbing
         for key, value in config_values.items():
             local_config[key] = value
         write_yaml(local_config, CONFIG_PATH)
+    # Install the default configuration
+    else:
+        local_config = configure_from_default()
```

Comment:
This line of code is going to be hard to understand long-term - to me, `configure_from_default` does not intuitively suggest that it will read the existing config.yaml if it exists. 

I would either rename `configure_from_default()` to clarify that it will read the current config.yaml if it exists, or expand the comment on line 208.

Reply:
Good point. I moved the check for an existing config into api._config. Now `configure_from_default` is equivalent to `configure_from_url`. I would have removed `configure_from_default` all together except for the failure case when we're unable to connect to the default stack.